docs/responses/body.md

```diff
@@ -95,6 +95,24 @@ basicRequest
 
 A number of JSON libraries are supported out-of-the-box, see [json support](../json.md).
 
+Sometimes it might be useful to model some http error responses right away. We can do that using `asTypedErrors` method:
+```scala
+case class SuccessModel(...)
+sealed trait MyErrorModel
+case class Conflict(...) extends MyErrorModel
+case class ServerError(...) extends MyErrorModel
+
+val myRequest: Request[Either[ResponseErrorTyped[MyErrorModel, io.circe.Error], MyModel], Nothing] =
+  basicRequest
+    .get(uri"https://example.com")
+    .response(asTypedError(asJsonAlways[SuccessModel]){
+      case StatusCode.Conflict => asJsonAlways[Conflict]
+      case StatusCode.ServerError => asJsonAlways[ServerError]
+    })
```

Comment:
now I'm thinking, that while it's good to have the logic centralised in `asTypedError`, maybe it would be more user-friendly to have a `asJsonEither` method as well.

Variant which uses different models for 2xx and different for all others:

```
basicRequest
     .get(uri"https://example.com")
     .response(asJsonEither[MyErrorModel, MyModel])
```

And the more general:

```
basicRequest
     .get(uri"https://example.com")
     .response(asJsonEitherFromStatusCode[MyErrorModel, MyModel]({
       case StatusCode.Conflict => asJsonAlways[Conflict]
       case StatusCode.ServerError => asJsonAlways[ServerError]
     })
```

what do you think>

Reply:
How about:
```
basicRequest
     .get(uri"https://example.com")
     .response(either(asJson[MyErrorModel], asJson[MyModel]))
```
and 
```
basicRequest
     .get(uri"https://example.com")
     .response(either(fromStatusCodes[MyErrorModel]{
       case StatusCode.Conflict => asJsonAlways[Conflict]
       case StatusCode.ServerError => asJsonAlways[ServerError]
     }), asJson[MyModel])
```

While I haven't thought even for a second whether it is possible, I like this structure and it might give additional flexibility since with it we should be able to parse errors from different formats than the main case.

Reply:
Yes, that's better :)